### ISSUE TITLE
add support for tags and check when deploying using nomad.

### ISSUE THREAD

**Issue by prabirshrestha** [hashicorp/waypoint#1125]
I'm using traefik to load balance http in nomad as mentioned here https://learn.hashicorp.com/tutorials/nomad/load-balancing-traefik

```hcl
    deploy {
        use "nomad" {
            tags = [
                "traefik.enable" = "true"
                "treafik.http.routes.http.rule=Path(`/myapp`)"
            ]
           check {
              type     = "http"
              path     = "/"
              interval = "2s"
              timeout  = "2s"
          }
        }
    }
```

This is very similar to docker https://github.com/hashicorp/waypoint/pull/1065

```
    service {
      name = "demo-webapp"
      port = "http"

      tags = [
        "traefik.enable=true",
        "traefik.http.routers.http.rule=Path(`/myapp`)",
      ]

      check {
        type     = "http"
        path     = "/"
        interval = "2s"
        timeout  = "2s"
      }
    }
```

**Comment by prabirshrestha** [comment]
There seems to be similar request here from multiple users.

https://github.com/hashicorp/waypoint/issues/567#issuecomment-710987735

**Comment by rugwirobaker** [comment]
Why does it seem like the release stanza is just a kubernetes feature? Shouldn't these tags be specified in the release section so that whatever is being used for service discovery picks them up? For example: 
https://github.com/hashicorp/waypoint/issues/567#issuecomment-787055263

**Comment by prabirshrestha** [comment]
For release to work waypoint url service would need to be configured. https://www.waypointproject.io/docs/url

I'm working on a mini PaaS called [zcloud](https://github.com/prabirshrestha/zcloud) based on hashicorp stack in the [hashicorp](https://github.com/prabirshrestha/zcloud/tree/hashicorp) branch except for proxies where I use traefik.  Seems like either we need the traefik url service provider for release or support tags in deploy for nomad.

**Comment by rugwirobaker** [comment]
I am also working on an internal PaaS that will run on top of nomad. This is probably the last missing piece for a working solution.

**Comment by briancain** [comment]
Hey all! Thanks for all of the extra context here. This is just a heads up that we have plans to support https://github.com/hashicorp/waypoint/issues/516 and other similar issues like this one in the 0.4 milestone around a feature we're calling mutable deployments. For now I'll be updating the milestone, but I wanted to check in to let you all know we are currently working on a good solution for this and haven't forgot about this 😄 thanks!

**Comment by mitchellh** [comment]
Fixed by #1299
